### ISSUE TITLE
Cassandra metrics change starting with version 1.2

### ISSUE THREAD

**Issue by alq666** [DataDog/jmxfetch#6]
Cassandra now exposes both:

```
org.apache.cassandra.db.ColumnFamilies <<<<<<<<<<< old
org.apache.cassandra.metrics.ColumnFamily <<<<<<<<<<< new
```

See [this](https://wiki.apache.org/cassandra/Metrics) for more details on the new style.

One key change is that the old format was:
1. CF is an MBean
2. CF has a bunch of attributes that map to metrics
while the new format has
1. CF is a scope
2. the MBean is the metric name
3. the attribute is named "Value"

**Comment by sethrosenblum** [comment]
I investigated this and it's going to require some changes to the way we collect tags from Cassandra.  Here's an example bean:

`Bean name: org.apache.cassandra.metrics:type=ColumnFamily,keyspace=Intake,scope=api_log,name=PendingTasks - Attribute name: Value - Attribute type: java.lang.Object`

The metric that we'd want from this would be along the lines of: `cassandra.pending_tasks` with tags: `keyspace:intake`, `columnfamily:api_log`

But jmxfetch turns this into: `cassandra.metrics.value` with tags: `name:pendingtasks` `keyspace:intake` `type:columnfamily` `scope:api_log`

As a result we'd have to add some function for Cassandra metrics (or make the jmx config extensible) to allow the jmx parser to create metric names like `cassandra.$name` and tags like `$type:$scope`.

We also wouldn't have good mechanisms for filtering the metrics that are collected, as we can't filter by the name attribute.

The deprecated metrics are going to be dropped in Cassandra 3.0.

**Comment by remh** [comment]
could probably be just a new version of https://github.com/DataDog/jmxfetch/blob/master/src/main/java/org/datadog/jmxfetch/reporter/Reporter.java#L105-L107

**Comment by jippi** [comment]
any update on this? Currently the cassandra jmx is more or less broken :(

**Comment by olivielpeau** [comment]
Fixed with #79, closing.
